views/js/pciCreator/dev/audioRecordingInteraction/runtime/js/providers/webAudio.js

```diff
@@ -164,12 +170,27 @@ define([
              * Close the audio context and destroy created assets
              */
             destroy: function destroy() {
+                var promises = [
+                    new Promise(function(resolve) {
+                        recorderWorker.terminate();
```

Comment:
the 1st part is synchronous, the promise is not needed. And since `audioContext.close()` already return a Promise, you can use that one (promises chain).

```js 
recorderWorker.terminate();
recorderWorker = null;
if(audioContext){
    return audioContext.close().then(function(){
          audioNodes = {};
      });
}